src/SNPN/Common/NetworkService.cs

```diff
@@ -154,6 +179,44 @@ public async Task<ResponseResult> SendNotification(QueuedNotificationItem notifi
 			});
 		}
 
+		public async Task<ResponseResult> SendNotificationFCM(QueuedNotificationItem notification)
+		{
+			return await _retryPolicy.ExecuteAsync(async () =>
+			{
+				_logger.Information("SendNotificationFCM {notificationUri}", notification.Uri);
+				var message = new Message()
+				{
+					Data = new Dictionary<string, string>()
+					{
+					["type"] = notification.MatchType.ToString(),
+					["username"] = notification.Post.Author,
+					["title"] = notification.Title,
+					["text"] = notification.Message,
+					["nlsid"] = notification.Post.Id.ToString(),
+					["parentid"] = notification.Post.ParentId.ToString(),
+					},
+					Token = notification.Uri.Replace("fcm://", ""),
+				};
+				var messaging = FirebaseMessaging.DefaultInstance;
+				try {
+					var result = await messaging.SendAsync(message);					
+					_logger.Information("SendNotificationFCM result: {result}", result);
+				}
+				catch (FirebaseMessagingException e) {
+					_logger.Error("SendNotificationFCM FirebaseMessagingException when trying to send to {notificationUri} {ErrorCode}", notification.Uri, e.ErrorCode);
+					if (e.ErrorCode == ErrorCode.NotFound) {
+						_logger.Warning("Removing FCM device with Uri {notificationUri} from DB", notification.Uri);
```

Comment:
This error can happen when a user uninstalls and re-installs ShackBrowse. This invalidates the FCM ID. So in the interest of not having the device list grow out of control, handle this error and remove the device.